packages/auth/src/grant/service.ts

```diff
@@ -0,0 +1,144 @@
+import { v4 } from 'uuid'
+import * as crypto from 'crypto'
+import { Transaction, TransactionOrKnex } from 'objection'
+
+import { BaseService } from '../shared/baseService'
+import { Grant, GrantState, StartMethod, FinishMethod } from './model'
+import { AccessRequest, isAccessRequest } from '../access/types'
+import { ClientInfo } from '../client/service'
+import { IAppConfig } from '../config/app'
+import { AccessService } from '../access/service'
+
+export interface GrantService {
+  validateGrantRequest(grantRequest: GrantRequest): boolean
```

Comment:
What do you think about moving this to the `GrantRoutes` (until it's removed for #351)?